types/block.go

```diff
@@ -1039,124 +1027,17 @@ type Data struct {
 
 // Hash returns the hash of the data
 func (data *Data) Hash() tmbytes.HexBytes {
-	if data.hash != nil {
-		return data.hash
-	}
-
-	// compute the data availability header
-	// todo(evan): add the non redundant shares back into the header
-	shares, _, err := data.ComputeShares(data.OriginalSquareSize)
-	if err != nil {
-		// todo(evan): see if we can get rid of this panic
-		panic(err)
+	if data == nil {
+		return (Txs{}).Hash()
 	}
-	rawShares := shares.RawShares()
-
-	eds, err := da.ExtendShares(data.OriginalSquareSize, rawShares)
-	if err != nil {
-		panic(err)
+	if data.hash == nil {
+		data.hash = data.Txs.Hash() // NOTE: leaves of merkle tree are TxIDs
 	}
 
-	dah := da.NewDataAvailabilityHeader(eds)
-
-	data.hash = dah.Hash()
-
+	// data.hash should be set by celestia-app in PrepareProposal
```

Comment:
[blocking]
can we make this comment clearer and either in the docs of this method, or at the top? I think being super clear about why we're doing both is important. We want tests to pass that don't use the app, so we also include normal way calculating the data hash.